### PR TITLE
HUE-9680 [Django Upgrade] upgrading Django 1.11.29 to 2.0.13

### PR DESCRIPTION
**Pull request by agl29**

### CODE REVIEW

**Comment by agl29** [comment]
5 tests are failing as of now, https://app.circleci.com/pipelines/github/cloudera/hue/4253/workflows/d72a0c80-6329-404a-94c5-1d60adaa1a0a/jobs/5795 .  In these tests there are some small-small mistakes are there like decoding or converting into str etc. Please take a look and guide me. @romainr

**Comment by romainr** [comment]
#1
`FAIL: desktop.models_tests.TestDocument2ImportExport.test_import_owned_document` should be the easiest as the `loaddata` command changed.

#2
The other ones not obvious. It feels like a Py2 vs Py3 conversion of the query handles HiveServerQueryHandle https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/models.py#L540 but here we should only have changed Django. Would recommend to debug one step by step and see why the upgrade leaked there. Maybe fixing 1. could help too.

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py [hunk 1]
```diff
@@ -92,8 +92,8 @@ def test_open_session(self):
         Session.objects.filter(owner=self.user, application=self.query_server['server_name']).count()
       )
       assert_equal(
-        session.guid,
-        Session.objects.get_session(self.user, self.query_server['server_name']).guid.encode()
+        str(session.guid),
+        Session.objects.get_session(self.user, self.query_server['server_name']).guid
```
Comment (agl29) [inline apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:96]: please suggest the right way for this @romainr

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py [hunk 2]
```diff
@@ -298,7 +298,7 @@ def test_get_table_with_error(self):
         client.get_table(database='database', table_name='table_name')
       except QueryServerException as e:
         if sys.version_info[0] > 2:
-          req_string = "TGetTablesReq(sessionHandle=TSessionHandle(sessionId=THandleIdentifier(guid=b'1', secret=b'1')), catalogName=None, schemaName='database', tableName='table_name', tableTypes=None)"
+          req_string = "TGetTablesReq(sessionHandle=TSessionHandle(sessionId=THandleIdentifier(guid=b'l\\xc4', secret=b'l\\xc4')), catalogName=None, schemaName='database', tableName='table_name', tableTypes=None)"
         else:
```
Comment (agl29) [inline apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:303]: for this also @romainr

Comment (romainr) [reply apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:303]: If you type Hive or Impala queries in a live Hue, does it work?

It feels like we double encode

File: desktop/core/requirements.txt [hunk 3]
```diff
@@ -9,7 +9,7 @@ channels-redis==2.3.2
 configobj==5.0.6
 cryptography==3.2
 django-auth-ldap==1.3.0
-Django==1.11.29 # Django 2 then 3?
+Django==2.0.13 # Django 2 then 3?
```
Comment (romainr) [inline desktop/core/requirements.txt:12]: How about we split the PR in two? (to not mix and match)

We do this line change and the failing Hive tests in a follow-up and can focus there?

**Comment by romainr** [comment]
How about better name for `upgrading django-nose for handling commit option to loaddata in Djang… …` and pushing it first?

File: desktop/core/src/desktop/api2.py [hunk 4]
```diff
@@ -860,7 +864,7 @@ def is_reserved_directory(doc):
   stdout = string_io()
   try:
     with transaction.atomic(): # We wrap both commands to commit loaddata & sync
-      management.call_command('loaddata', f.name, verbosity=3, traceback=True, stdout=stdout, commit=False) # We need to use commit=False because commit=True will close the connection and make Document.objects.sync fail.
+      management.call_command('loaddata', f.name, verbosity=3, traceback=True, stdout=stdout)
```
Comment (romainr) [inline desktop/core/src/desktop/api2.py:867]: Is there an equivalent for doing commit=False?

Comment (agl29) [reply desktop/core/src/desktop/api2.py:867]: https://github.com/jazzband/django-nose/pull/308 This PR is related to this. I think this the right way of doing this. @romainr

Comment (romainr) [reply desktop/core/src/desktop/api2.py:867]: :+1:

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py (hunk 2 above)
Comment (agl29) [reply apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:303]: yes, with this branch locally, Hive queries are working.
Screenshot 2021-01-12 at 9 24 03 PM

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py (hunk 1 above)
Comment (agl29) [reply apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:96]: tracking @ https://github.com/cloudera/hue/pull/1621

File: apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py (hunk 2 above)
Comment (agl29) [reply apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py:303]: tracking @ https://github.com/cloudera/hue/pull/1621

File: desktop/core/src/desktop/api2.py (hunk 4 above)
Comment (agl29) [reply desktop/core/src/desktop/api2.py:867]: tracking @ https://github.com/cloudera/hue/pull/1652

**Review by romainr — APPROVED** [review]
Nice!

**Pull request merged**